test/hotspot/jtreg/compiler/lib/verify/Verify.java

```diff
@@ -71,107 +109,176 @@ private static void checkEQ(Object a, Object b, String context) {
+            case Exception x -> checkEQimpl(x, (Exception) b,               field, aParent, bParent);
             default -> {
-                System.err.println("ERROR: Verify.checkEQ failed: type not supported: " + ca.getName());
-                print(a, "a " + context);
-                print(b, "b " + context);
-                throw new VerifyException("Object array type not supported: " + ca.getName());
+                if (ca.getName().startsWith("jdk.incubator.vector") && ca.getName().contains("Vector")) {
+                    // We do not want to import jdk.incubator.vector explicitly, because it would mean we would also have
+                    // to add "--add-modules=jdk.incubator.vector" to the command-line of every test that uses the Verify
+                    // class. So we hack this via reflection.
+                    Object va = null;
+                    Object vb = null;
+                    try {
+                        Method m = ca.getMethod("toArray");
+                        m.setAccessible(true);
+                        va = m.invoke(a);
+                        vb = m.invoke(b);
+                    } catch (NoSuchMethodException e) {
+                        throw new RuntimeException("Could not invoke toArray on " + ca.getName(), e);
+                    } catch (IllegalAccessException e) {
+                        throw new RuntimeException("Could not invoke toArray on " + ca.getName(), e);
+                    } catch (InvocationTargetException e) {
+                        throw new RuntimeException("Could not invoke toArray on " + ca.getName(), e);
+                    }
+                    checkEQdispatch(va, vb, field + ".toArray", aParent, bParent);
+                    return;
+                }
+
+                checkEQArbitraryClasses(a, b);
+                return;
             }
         }
     }
 
     /**
      * Verify that two bytes are identical.
      */
-    private static void checkEQimpl(byte a, byte b, String context) {
+    private void checkEQimpl(byte a, byte b, String field, Object aParent, Object bParent) {
         if (a != b) {
-            System.err.println("ERROR: Verify.checkEQ failed: value mismatch: " + a + " vs " + b + " for " + context);
+            System.err.println("ERROR: Verify.checkEQ failed: value mismatch: " + a + " vs " + b);
+            print(a, b, field, aParent, bParent);
             throw new VerifyException("Value mismatch: " + a + " vs " + b);
         }
     }
 
     /**
      * Verify that two chars are identical.
      */
-    private static void checkEQimpl(char a, char b, String context) {
+    private void checkEQimpl(char a, char b, String field, Object aParent, Object bParent) {
         if (a != b) {
-            System.err.println("ERROR: Verify.checkEQ failed: value mismatch: " + (int)a + " vs " + (int)b + " for " + context);
+            System.err.println("ERROR: Verify.checkEQ failed: value mismatch: " + (int)a + " vs " + (int)b);
```

Comment:
Why do you need an upcast here? Same for `short`.

Reply:
Look at this ;)
```
jshell> char a = 66;
a ==> 'B'

jshell> System.out.println("a: " + a);
a: B

jshell> System.out.println("a: " + (int)a);
a: 66
```

But I can remove the casts for `short`.

Reply:
Added a comment as well.

Reply:
Right, that makes sense for the `char` case. But good that we could remove it for the `short` case.